### ISSUE TITLE
Escaped terms in filter field raises exception

### ISSUE THREAD

**Issue by magedan** [sonata-project/SonataAdminSearchBundle#5]
Input of special characters/terms (like + - && || ! etc) as a string filter value causes following error:

> The produced query is not a valid json string : "{"match":{"name":{"query":"test-hyphen","operator":"and"}}}"
> 500 Internal Server Error - InvalidException

This is because of the \Elastica\Util::escapeTerm() call in Sonata\AdminSearchBundle\Filter\StringFilter:

``` php
        $queryBuilder
            ->fieldOpen($secondOperator)
                ->fieldOpen($field)
                    ->field('query', Util::escapeTerm($data['value']))
                    ->field('operator', 'and')
                ->fieldClose()
            ->fieldClose();
```

If I remove Util::escapeTerm() call Elastica works as expected, here is the full query:

``` javascript
{"query":{"bool":{"must":[{"match":{"name":{"query":"test-hyphen","operator":"and"}}}]}},"sort":{"id":{"order":"desc"}},"from":0,"size":32}
```

Is this really necessary to apply escapeTerm here?

**Comment by greg0ire** [comment]
I am away right now, but iirc, you will find the answer in the git history

**Comment by greg0ire** [comment]
Fixed by 836b19f4d212ebf2025bbc03637d47377bdc6572 , thank you @magedan
